core/logger/logger.go

```diff
@@ -43,7 +43,7 @@ func init() {
 //
 // Levels
 //  - Fatal: Logs and then calls os.Exit(1). Be careful about using this since it does NOT unwind the stack and may exit uncleanly.
-//  - Panic: Unrecoverable error. Example: invariant violation, programmer error
+//  - Panic: Unrecoverable error. Example: AssumptionViolation, programmer error
```

Comment:
Did we come to a conclusion about these? Why replace a well known idiomatic term here?

Reply:
I thought the logging conclusion was to log AssumptionViolation and document that everywhere?

Reply:
I'm just going to remove that commit from this PR to avoid any unrelated controversy